learn/shorts/intro-gh-actions.qmd

```diff
@@ -0,0 +1,369 @@
+3.  Create a new file in the `.github/workflows/` folder called
+    `build-website.yml`.
+4.  Open the file in a text editor and start writing out these "code"
+    instructions.
+
+We'll start with the `on` section, as well as the `name` of the Action.
+In this case, we want the Action to always run whenever you or someone
+else pushes changes to the `main` branch of the repository. So we would
+start with:
+
+``` yaml
+name: Build website
+on:
+  push:
+    branches:
+      - main
+```
+
+Because we want to GitHub to host the website on their GitHub Pages
+service, which uses a special branch called `gh-pages` in the
+repository, we need to give the Action some permissions to be able to
+write to that branch. We'll add that by using the `permissions` section:
+
+``` yaml
+permissions: 
+    contents: write
+    pages: write
+```
+
+Next is the core part of the Action, the `jobs` section. Because we want
+to build the contents of the repository using Quarto and then push to
+the `gh-pages` branch, we need at least three steps:
+
+1.  Downloading the contents of the repository into the GitHub server by
+    checking out the repository.
+2.  Installing Quarto in the server.
+3.  Running Quarto on the repository contents to build the website and
+    store the output in the branch called `gh-pages`.
+
+Thankfully, we don't have to do all of these steps manually, since there
+already exists Actions that you can download and use, such as the
+`checkout` action provided by GitHub's `actions` account, along with the
+`quarto-actions/setup` and `quarto-actions/publish` actions provided by
+the Quarto team in their `quarto-dev` GitHub account.
+
+So we'll first set up the `jobs` section with the `name` and to
+`runs-on` the latest version of Ubuntu:
+
+``` yaml
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+```
+
+Next is to start the `steps` for the job, first of which is to check out
+the repository:
+
+``` yaml
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+```
+
+Note the use of `@4` to indicate the version of the `checkout` action to
+use. Next, we'll install Quarto:
+
+``` yaml
+      - name: Set up Quarto
+        uses: quarto-dev/quarto-actions/setup@v2
+```
+
+In this case, we need to provide the Quarto `setup` action (using
+version `v2`). Lastly, we want to run Quarto to build the website and
+publish it to the `gh-pages` branch:
+
+``` yaml
+      - name: Publish to GitHub Pages (and render) 
+        uses: quarto-dev/quarto-actions/publish@v2
+        with:
+          target: gh-pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+```
+
+Because we will be saving the built website to the `gh-pages` branch, we
+need to specify the output location (using the `target` argument
+accessible with the `with:` field) to be `gh-pages`. Saving to the
+`gh-pages` branch requires using the `GITHUB_TOKEN` environment, which
+is found within the `secrets` object.
+
```

Comment:
I find it very helpful to see scripts like this in their entirety. 

```suggestion

The action now look like this in it's entirety: 

``` yaml
name: Build website
on:
  push:
    branches:
      - main
permissions: 
    contents: write
    pages: write
jobs:
  build-deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Check out repository
        uses: actions/checkout@v4
        - name: Set up Quarto
        uses: quarto-dev/quarto-actions/setup@v2
        - name: Publish to GitHub Pages (and render) 
        uses: quarto-dev/quarto-actions/publish@v2
        with:
          target: gh-pages
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 

```
Again, bc of the backticks used for both the yaml block and the suggestion block, this suggestion is missing ` ``` ` in the end.